File: lib/array/arrTo1dArray.ts
import { is2dArray } from '../verify/is2dArray.js';

export function arrTo1dArray(array: any[]): any[] {
	const convert1dArray = (arr: any[]): any[] => {
		const tempArr = [];
		const arrayLength = arr.length;

		for (let i = 0; i < arrayLength; i += 1) {
			if (typeof arr[i] !== 'object') {
				tempArr.push(arr[i]);
			} else if (is2dArray(arr[i])) {
				tempArr.push(...convert1dArray(arr[i]));
			} else {
				tempArr.push(...arr[i]);
			}
		}

		return tempArr;
	};

	return convert1dArray(array);
}

File: packages/python/qsu/array/arrTo1dArray.py
from typing import Sequence, cast

from ..verify.is2dArray import is2dArray


def arrTo1dArray(array: list) -> list:
	def convert1dArray(arr: Sequence) -> list:
		tempArr: list = []

		for item in arr:
			if not isinstance(item, (list, tuple)):
				tempArr.append(item)
			elif is2dArray(cast(list, item)):
				tempArr.extend(convert1dArray(item))
			else:
				tempArr.extend(item)

		return tempArr

	return convert1dArray(array)
